src/features/creator-request/server.ts

```
@@ -0,0 +1,168 @@
+} from "@/contracts/creator-request.contract";
+import { db } from "@/db";
+import { creatorRequest, user } from "@/db/schema";
+import { getAuthSession } from "@/lib/auth-actions";
+import { requireAdmin, requireMember } from "@/lib/auth-middleware";
+
+export const submitRequest = createServerFn({ method: "POST" })
+  .middleware([requireMember])
+  .inputValidator(submitCreatorRequestContract)
+  .handler(async ({ data, context }) => {
+    const newRequest = {
+      userId: context.user.id,
+      portfolioLink: data.portfolioLink,
+      motivation: data.motivation,
+    };
+
+    const result = await db
+      .insert(creatorRequest)
+      .values(newRequest)
+      .returning({ id: creatorRequest.id });
+
+    return result[0];
+  });
+
+export const approveRequest = createServerFn({ method: "POST" })
+  .middleware([requireAdmin])
+  .inputValidator(approveCreatorRequestContract)
+  .handler(async ({ data, context }) => {
+    const { userId, requestId, note } = data;
+    try {
+      await db.update(user).set({ role: "creator" }).where(eq(user.id, userId)).returning();
+      await db
+        .update(creatorRequest)
+        .set({ status: "approved", reviewedBy: context.user.id, note })
+        .where(eq(creatorRequest.id, requestId));
+    } catch (error) {
+      if (error instanceof Error) console.log(error.message);
+    }
+  });
+
+export const rejectRequest = createServerFn({ method: "POST" })
+  .middleware([requireAdmin])
+  .inputValidator(rejectCreatorRequestContract)
+  .handler(async ({ data, context }) => {
+    const { requestId } = data;
+    try {
+      await db
+        .update(creatorRequest)
+        .set({ status: "rejected", reviewedBy: context.user.id })
+        .where(eq(creatorRequest.id, requestId));
+    } catch (error) {
+      if (error instanceof Error) console.log("Creator request rejection has failed");
+    }
+  });
+
+export const listAllPendingRequests = createServerFn({ method: "GET" })
```

The list functions (`listAllPendingRequests`, `listOwnRequests`, etc.) catch errors and log them to console, but implicitly returning `undefined`. This swallows exceptions that should likely be handled by the UI. If the database query fails, the frontend receives undefined and renders an empty list, leading the admin/user to believe there are no requests rather than realizing the system is broken.

Suggestion: remove the try/catch block in the server functions and let the error propagate to the client, or return a structured result object (e.g., { success: false, error: ... }) that the client can handle explicitly.